services/template_watch/template_watch.py

```diff
@@ -0,0 +1,46 @@
+import os
+from os.path import join as pjoin
+import time
+import sys
+
+from watchdog.observers import Observer
+from watchdog.events import PatternMatchingEventHandler
+
+from baselayer.log import make_log
+from baselayer.tools.fill_conf_values import fill_config_file_values
+from baselayer.app.env import load_env
+
+log = make_log('template_watch')
+env, cfg = load_env()
+
+if not env.debug:
+    log('Watcher only used in debug mode; exiting')
+    sys.exit(0)
+
+# Baselayer parent project path
+watchpath = os.path.abspath(pjoin(os.path.dirname(__file__), '../../..'))
+
+
+class EventHandler(PatternMatchingEventHandler):
+    def __init__(self):
+        PatternMatchingEventHandler.__init__(self, patterns=['*.template'])
+
+    def on_modified(self, event):
+        fill_config_file_values([event.src_path])
```

Comment:
Maybe log which template changed and triggered this?

Reply:
It's already logged by the template compiler.